pkg/manager/member/tidb_member_manager.go

```diff
@@ -283,37 +283,54 @@ func (m *tidbMemberManager) syncTiDBService(tc *v1alpha1.TidbCluster) error {
 		return fmt.Errorf("syncTiDBService: failed to get svc %s for cluster %s/%s, error: %s", newSvc.Name, ns, tc.GetName(), err)
 	}
 	oldSvc := oldSvcTmp.DeepCopy()
+	if newSvc.Annotations == nil {
+		newSvc.Annotations = map[string]string{}
+	}
+	if oldSvc.Annotations == nil {
+		oldSvc.Annotations = map[string]string{}
+	}
+	if newSvc.Labels == nil {
+		newSvc.Labels = map[string]string{}
+	}
+	if oldSvc.Labels == nil {
+		oldSvc.Labels = map[string]string{}
+	}
 	util.RetainManagedFields(newSvc, oldSvc)
 
 	equal, err := controller.ServiceEqual(newSvc, oldSvc)
 	if err != nil {
 		return err
 	}
 	annoEqual := util.IsSubMapOf(newSvc.Annotations, oldSvc.Annotations)
+	labelEqual := util.IsSubMapOf(newSvc.Labels, oldSvc.Labels)
 	isOrphan := metav1.GetControllerOf(oldSvc) == nil
 
-	if !equal || !annoEqual || isOrphan {
-		svc := *oldSvc
-		svc.Spec = newSvc.Spec
-		err = controller.SetServiceLastAppliedConfigAnnotation(&svc)
-		if err != nil {
-			return err
-		}
-		svc.Spec.ClusterIP = oldSvc.Spec.ClusterIP
-		// apply change of annotations if any
-		for k, v := range newSvc.Annotations {
-			svc.Annotations[k] = v
-		}
-		// also override labels when adopt orphan
-		if isOrphan {
-			svc.OwnerReferences = newSvc.OwnerReferences
-			svc.Labels = newSvc.Labels
-		}
-		_, err = m.deps.ServiceControl.UpdateService(tc, &svc)
-		return err
+	if equal && annoEqual && labelEqual && !isOrphan {
+		return nil
 	}
 
-	return nil
+	svc := *oldSvc
+	svc.Spec = newSvc.Spec
+	err = controller.SetServiceLastAppliedConfigAnnotation(&svc)
+	if err != nil {
+		return err
+	}
+	svc.Spec.ClusterIP = oldSvc.Spec.ClusterIP
+	// apply change of annotations if any
+	for k, v := range newSvc.Annotations {
+		svc.Annotations[k] = v
+	}
+	// apply change of labels if any
+	for k, v := range newSvc.Labels {
+		svc.Labels[k] = v
+	}
+	// also override labels when adopt orphan
+	if isOrphan {
+		svc.OwnerReferences = newSvc.OwnerReferences
+		svc.Labels = removeHelmLabels(svc.Labels)
```

Comment:
Why helm labels should be removed?

Reply:
After migration from helm to operator, helm labels are useless. Unnecessary labels could interfere with users. Some existing unit tests fails if helm labels not removed.

So I remove helm labels explicitly here.

Reply:
Which case failed? Why it did not fail before?

Reply:
Why helm labels are deleted by the operator if they are not patched by the operator?

How to deal with other unknown labels? I think helm labels are not special.

Reply:
the origin code 's behavier is ignore the old labels
<img width="539" alt="Screen Shot 2021-05-21 at 5 42 24 PM" src="https://user-images.githubusercontent.com/1681864/119117541-e9aab080-ba5b-11eb-9060-9f90abcb3f8e.png">

Reply:
if we want to keep the old labels we must set them in `tc` since we have adopt the sts.
I will keep the behavior for now